flexget/plugins/api_trakt.py

```diff
@@ -147,136 +487,219 @@ class TraktSearchResult(Base):
+    }
+    if style == 'show':
+        ids['tvdb_id'] = tvdb_id
+        ids['tvrage_id'] = tvrage_id
+        model = TraktShow
+    else:
+        model = TraktMovie
+    result = None
+    if any(ids.values()):
+        result = session.query(model).filter(
+            or_(getattr(model, col) == val for col, val in ids.iteritems() if val)).first()
+    elif title:
+        title, y = split_title_year(title)
+        year = year or y
+        query = session.query(model).filter(model.title == title)
+        if year:
+            query = query.filter(model.year == year)
+        result = query.first()
+    return result
+
+
+def get_trakt(style=None, title=None, year=None, trakt_id=None, trakt_slug=None, tmdb_id=None, imdb_id=None, tvdb_id=None,
+              tvrage_id=None):
+    """Returns the matching media object from trakt api."""
+    # TODO: Better error messages
+    # Trakt api accepts either id or slug (there is a rare possibility for conflict though, e.g. 24)
+    trakt_id = trakt_id or trakt_slug
+    req_session = get_session()
+    if not trakt_id:
+        # Try finding trakt_id based on other ids
+        ids = {
+            'imdb': imdb_id,
+            'tmdb': tmdb_id
+        }
+        if style == 'show':
+            ids['tvdb'] = tvdb_id
+            ids['tvrage'] = tvrage_id
+        for id_type, identifier in ids.iteritems():
+            if not identifier:
+                continue
+            try:
+                results = req_session.get(get_api_url('search'), params={'id_type': id_type, 'id': identifier}).json()
+            except requests.RequestException as e:
+                log.debug('Error searching for trakt id %s' % e)
+                continue
+            for result in results:
+                if result['type'] != style:
+                    continue
+                trakt_id = result[style]['ids']['trakt']
                 break
-    if not series:
-        raise LookupError('Could not find tvdb_id for `%s` on trakt' % title)
-    return series
+            if trakt_id:
+                break
+        if not trakt_id and title:
+            # Try finding trakt id based on title and year
+            title, y = split_title_year(title)
+            year = year or y
+            clean_title = normalize_series_name(title)
+            try:
+                results = req_session.get(get_api_url('search'), params={'query': clean_title, 'type': style}).json()
+            except requests.RequestException as e:
+                raise LookupError('Searching trakt for %s failed with error: %s' % (title, e))
+            for result in results:
+                if year and result[style]['year'] != year:
+                    continue
+                if clean_title == normalize_series_name(result[style]['title']):
+                    trakt_id = result[style]['ids']['trakt']
+                    break
+    if not trakt_id:
+        raise LookupError('Unable to find %s on trakt.' % type)
```

Comment:
If You don't have a trakt ID, IMDb, or tvdb then the only thing left is the title.